server/opts.go

```diff
@@ -1522,6 +1522,9 @@ func parseJetStream(v interface{}, opts *Options, errors *[]error, warnings *[]e
 				opts.JetStreamMaxStore = mv.(int64)
 			case "domain":
 				opts.JetStreamDomain = mv.(string)
+				if subj := fmt.Sprintf(jsDomainAPI, opts.JetStreamDomain); !IsValidSubject(subj) {
```

Comment:
In order to check for both file config and for users embedding the server, I recommend doing validation checks in server.go, `validateOptions()`

Reply:
Good point..